src/runtime/pyscope.cs

```diff
@@ -0,0 +1,535 @@
+        /// GetVariable Method
+        /// </summary>
+        /// <remarks>
+        /// Returns the value of the variable, local variable first.
+        /// If the variable is not exists, throw an Exception.
+        /// </remarks>
+        public PyObject GetVariable(string name)
+        {
+            Check();
+            using (var pyKey = new PyString(name))
+            {
+                if (Runtime.PyMapping_HasKey(variables, pyKey.obj) != 0)
+                {
+                    IntPtr op = Runtime.PyObject_GetItem(variables, pyKey.obj);
+                    if (op == IntPtr.Zero)
+                    {
+                        throw new PythonException();
+                    }
+                    if (op == Runtime.PyNone)
+                    {
+                        return null;
+                    }
+                    return new PyObject(op);
+                }
+                else
+                {
+                    throw new PyScopeException(String.Format("'ScopeStorage' object has no attribute '{0}'", name));
+                }
+            }
+        }
+
+        /// <summary>
+        /// TryGetVariable Method
+        /// </summary>
+        /// <remarks>
+        /// Returns the value of the variable, local variable first.
+        /// If the variable is not exists, return null.
+        /// </remarks>
+        public bool TryGetVariable(string name, out PyObject value)
+        {
+            Check();
+            using (var pyKey = new PyString(name))
+            {
+                if (Runtime.PyMapping_HasKey(variables, pyKey.obj) != 0)
+                {
+                    IntPtr op = Runtime.PyObject_GetItem(variables, pyKey.obj);
+                    if (op == IntPtr.Zero)
+                    {
+                        throw new PythonException();
+                    }
+                    if (op == Runtime.PyNone)
+                    {
+                        value = null;
+                        return true;
+                    }
+                    value = new PyObject(op);
+                    return true;
+                }
+                else
+                {
+                    value = null;
+                    return false;
+                }
+            }
+        }
+
+        public T GetVariable<T>(string name)
+        {
+            Check();
+            PyObject pyObj = GetVariable(name);
+            if (pyObj == null)
+            {
+                return default(T);
+            }
+            return pyObj.As<T>();
+        }
+
+        public bool TryGetVariable<T>(string name, out T value)
+        {
+            Check();
+            PyObject pyObj;
+            var result = TryGetVariable(name, out pyObj);
+            if (!result)
+            {
+                value = default(T);
+                return false;
+            }
+            if (pyObj == null)
+            {
+                value = default(T);
+                return true;
+            }
```

Comment:
This means that if someone does `TryGetVariable<int>("something", out val)` where `something = None`, they will get as a result `val = 0` without any further indication. Seems like a very bad idea for value types.